### PR TITLE
Remove travis OSX build

### PR DESCRIPTION
**Pull request by RX14**
Travis's OSX infastructure is extremely overloaded. We think that the excess of OSX jobs is filling up our travis "build slots" and causing linux builds not to run. To remedy this we've recided to remove OSX from travis, as we already have circleci.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,32 +1,40 @@
...
-sudo: required
+
 dist: trusty
+sudo: required
+
 services:
   - docker
+
 matrix:
-  fast_finish: true
```
Comment (oprypin) [inline .travis.yml]: There's no reason to remove `fast_finish`. We still have multiple builds and notifying the author as soon as one of them fails is still nice to have.

**Comment by oprypin** [comment]
I disagree with the big amount of spacing added. Not to mention that it's less clear what the change is.

File: .travis.yml (hunk 1 above)
Comment (RX14) [reply .travis.yml]: I read the fast_finish docs and it made it sound as if it was only valid with `allow_failures`.

Comment (oprypin) [reply .travis.yml]: With fast finishing enabled, Travis CI will mark your build as finished as soon as one of two conditions are met: The only remaining jobs are allowed to fail, **or a job has already failed**.

Comment (RX14) [reply .travis.yml]: Oops.

**Comment by RX14** [comment]
Why do you dislike the spacing? It's a lot clearer with whitespace between each section.

I also split it into 2 commits so the diff was clear.

File: .travis.yml (hunk 1 above)
Comment (RX14) [reply .travis.yml]: Fixed this BTW

**Comment by mverzilli** [comment]
Let's ping approved PR authors to rebase so they don't block the queues

**Comment by oprypin** [comment]
I don't think that's needed. Config is not taken from PRs, that's insecure

**Comment by RX14** [comment]
@oprypin so why did this PR not build OSX?

**Comment by oprypin** [comment]
Hm nevermind. But I still think everything will work out without manual rebasing. PRs get rebased anyway.

**Comment by RX14** [comment]
@oprypin but we want green CI before they're rebased onto master.

**Comment by oprypin** [comment]
So the problem is the PRs that failed the last time they were checked. Have you tried simply restarting the build in the Travis interface?

**Comment by RX14** [comment]
@oprypin no, this is about fixing PRs hanging in the "queueing" status for several days.

**Comment by mverzilli** [comment]
Restarting the build in Travis runs on the targets defined before this PR. But I think that's ok really. Let's see in a case by case basis, as needed.

**Pull request merged**